structuralcodes/materials/concrete/_concreteEC2_2023.py

```diff
@@ -247,9 +271,161 @@ def eps_cu1(self) -> float:
+
+        Args:
+            value (float): the new value for eps_c2, no units
+        """
+        if abs(value) >= 0.1:
+            warnings.warn(
+                'A suspect value is input for eps_c2 that should be a pure'
+                ' number without units. Plase check ({value} given).'
+            )
+        self._esp_c2 = value
+
+    @property
+    def eps_cu2(self) -> float:
+        """Returns the strain at concrete failure of concrete for the
+            Parabola-rectangle constitutive law.
+
+        Returns:
+            float: the maximum strain at failure of concrete
+        """
+        self._eps_cu2 = self._eps_cu2 or ec2_2023.eps_cu2()
+        return self._eps_cu2
+
+    @eps_cu2.setter
+    def eps_cu2(self, value: float):
+        """Sets the strain at concrete failure of concrete for the
+            Parabola-rectangle constitutive law.
+
+        Args:
+            value (float): the new value for eps_cu2, no units
+        """
+        if abs(value) >= 0.1:
+            warnings.warn(
+                'A suspect value is input for eps_cu2 that should be a pure'
+                ' number without units. Plase check ({value} given).'
+            )
+        self._esp_cu2 = value
+
+    @property
+    def n_parabolic_rectangular(self) -> float:
+        """Returns the coefficient for Parabola-rectangle constitutive law.
+
+        Returns:
+        float: the exponent for Parabola-recangle law.
+        """
+        self._n_parabolic_rectangular = (
+            self._n_parabolic_rectangular
+            or ec2_2023.n_parabolic_rectangular(self.fck)
+        )
+        return self._n_parabolic_rectangular
+
+    @n_parabolic_rectangular.setter
+    def n_parabolic_rectangular(self, value: float):
+        """Sets the coefficient for Parabola-rectangle constitutive law.
+
+        Args:
+            value (float): the new value for n, no units
+
+        Raises:
+            ValueError if value < 0
+        """
+        if value < 0:
+            raise ValueError(f'n should be a positive value ({value} given)')
+        if value >= 5:
+            warnings.warn(
+                'A suspect value is input for eps_cu2 that should be a pure'
+                ' number without units. Plase check ({value} given).'
+            )
+        self._n_parabolic_rectangular = value
+
     @property
     def gamma_c(self) -> float:
         """The partial factor for concrete."""
         # Here we should implement the interaction with the globally set
         # national annex. For now, we simply return the default value.
         return self._gamma_c or 1.5
+
+    def __elastic__(self) -> dict:
+        """Returns kwargs for creating an elastic constitutive law."""
+        return {'E': self.Ecm}
+
+    def __parabolarectangle__(self) -> dict:
+        """Returns kwargs for creating a parabola rectangle const law."""
+        return {'fc': self.fcd(), 'eps_0': -0.002, 'eps_u': -0.0035, 'n': 2}
```

Comment:
Here we should call the properties and not the default values.

Reply:
Done in 16967e2fe4a6290daab9551cb04ae7167cc69c21